src/lightcurvelynx/utils/extrapolate.py

```diff
@@ -279,37 +353,185 @@ def __init__(self, rate):
         """Evaluate the extrapolation given the last valid points(s) and a list of new
         query points.
 
         Note
         ----
         This function does not care which axis is being extrapolated. The returned values are
-        always len(query_values) x len(last_flux) and may need to be transposed by the calling
+        always len(query_values) x len(last_fluxes) and may need to be transposed by the calling
         function.
 
         Parameters
         ----------
-        last_value : float
-            The last valid value along the extrapolation axis a which the flux was predicted
+        last_values : float
+            The last valid value along the extrapolation axis at which the flux was predicted
             (e.g., wavelength in AA or time in days).
-        last_flux : numpy.ndarray
+        last_fluxes : numpy.ndarray
             A length N array of the flux values at the last valid time or wavelength (in nJy).
         query_values : numpy.ndarray
             A length M array of values along the extrapolation axis (times in days or wavelengths
             in AA) at which to extrapolate.
+        extrap_axis : str
+            Extrapolation axis. 'time' or 'wavelength'.
 
         Returns
         -------
         flux : numpy.ndarray
             A N x M matrix of extrapolated values. Where M is the number of query points and
             N is the number of flux values at the last valid point.
         """
-        last_flux = np.asarray(last_flux)
+
+        if extrap_axis not in ["time", "wavelength"]:
+            raise ValueError("Invalid extrap_axis value: only 'time' or 'wavelength' is valid")
+        if not np.isscalar(last_values):
+            if np.any(last_values > np.max(query_values)):
+                last_values = float(last_values[0])
+                if extrap_axis == "time":
+                    last_fluxes = last_fluxes[0, :]
+                else:
+                    last_fluxes = last_fluxes[:, 0]
+            else:
+                last_values = float(last_values[-1])
+                if extrap_axis == "time":
+                    last_fluxes = last_fluxes[-1, :]
+                else:
+                    last_fluxes = last_fluxes[:, -1]
+
+        last_fluxes = np.asarray(last_fluxes)
         query_values = np.asarray(query_values)
-        dist = np.abs(query_values - last_value)
+        dist = np.abs(query_values - last_values)
 
         multiplier = np.exp(-self.rate * dist)
-        flux = last_flux[:, np.newaxis] * multiplier[np.newaxis, :]
+        flux = last_fluxes[:, np.newaxis] * multiplier[np.newaxis, :]
+        return flux
+
+
+class LinearFit(FluxExtrapolationModel):
+    """Linear extrapolation based on a linear fit to the last few points.
+
+    Attributes
+    ----------
+    nfit : float
+        The number of points to perform the linear fit on.
+    """
+
+    def __init__(self, nfit=5):
+        self.nfit = nfit
```

Comment:
If we do make `nfit` an attribute of the base class, we would need to reverse the order of the two statements here in order to set a default and then override it.